docs/v1/examples/media.md

```diff
@@ -0,0 +1,215 @@
+      }
+    ]
+  }
+  else {
+    // User is browsing the website
+    // Only update browsing timestamp when changing from watching to browsing
+    if (wasWatchingVideo) {
+      browsingTimestamp = Math.floor(Date.now() / 1000)
+      wasWatchingVideo = false
+    }
+
+    presenceData.details = 'Browsing'
+    presenceData.state = 'Looking for videos'
+    presenceData.startTimestamp = browsingTimestamp
+  }
+
+  // Update wasWatchingVideo state for the next update
+  if (video && video.readyState > 0 && !video.paused) {
+    wasWatchingVideo = true
+  }
+
+  // Set the activity
+  if (presenceData.details) {
+    presence.setActivity(presenceData)
+  }
+  else {
+    presence.clearActivity()
+  }
+})
+```
+
+## How It Works
+
+1. We import the `getTimestampsFromMedia` and `Assets` from the `premid` package.
+2. We define an `ActivityAssets` enum for custom assets used in our activity.
+3. We create a new `Presence` instance with a client ID.
+4. We initialize a `browsingTimestamp` variable outside the UpdateData event to maintain consistent timing.
+5. We listen for the `UpdateData` event, which is fired regularly by the PreMiD extension.
+6. We use destructuring to access document.location properties: `const { pathname, hostname, href } = document.location`.
+7. We create a `PresenceData` object with a `largeImageKey` property using our ActivityAssets enum.
+8. We get the video element using `document.querySelector("video")`.
+9. If a video is found and it's ready to play:
+   - We get the video title and author from the page.
+   - We set the activity type to `ActivityType.Watching`.
+   - We set the details and state to show the video title and author.
+   - We set the large image text to the name of the service.
+   - If the video is playing:
+     - We set the small image key to `Assets.Play` to indicate that the video is playing.
+     - We calculate timestamps using destructuring: `[presenceData.startTimestamp, presenceData.endTimestamp] = getTimestampsFromMedia(video)`.
+   - If the video is paused:
+     - We set the small image key to `Assets.Pause` to indicate that the video is paused.
+   - We add buttons to link to the video and the channel.
+10. If no video is found or it's not ready to play:
+    - We only update the browsing timestamp when changing from watching to browsing.
+    - We set the details and state to indicate that the user is browsing the website.
+    - We use the consistent browsing timestamp to show how long the user has been browsing.
+11. We update the `wasWatchingVideo` state for the next update cycle.
+12. Finally, we set the activity using `presence.setActivity()`.
+
+## Handling Different Media Types
+
+You can modify this example to handle different types of media:
+
+### For Audio
+
+```typescript
+// Set the activity type to Listening
+presenceData.type = ActivityType.Listening
+
+// Set the details and state
+presenceData.details = 'Listening to'
+presenceData.state = `${title} by ${artist}`
```

Comment:
title and artist can be its own lines, details set to listening to is not correct, that activity type already does that.